src/workerd/util/tracing.c++

```diff
@@ -0,0 +1,97 @@
+#include "tracing.h"
+#include <kj/debug.h>
+#include <kj/filesystem.h>
+#include <kj/memory.h>
+#include <kj/vector.h>
+#include <fcntl.h>
+#include <perfetto/tracing/track_event_legacy.h>
+#include "protos/perfetto/config/data_source_config.gen.h"
+#include "protos/perfetto/config/trace_config.gen.h"
+#include "protos/perfetto/config/track_event/track_event_config.gen.h"
+
+PERFETTO_TRACK_EVENT_STATIC_STORAGE_IN_NAMESPACE(workerd);
+
+namespace workerd {
+namespace {
+kj::AutoCloseFd openTraceFile() {
+  int fd = open("/tmp/PERFETTO_TRACE", O_RDWR | O_CREAT | O_TRUNC, 0600);
+  KJ_REQUIRE(fd >= 0, "Unable to open tracing file");
+  return kj::AutoCloseFd(fd);
+}
+
+bool perfettoInitialized = false;
+void initializePerfettoOnce() {
+  if (perfettoInitialized) return;
+  perfettoInitialized = true;
+  perfetto::TracingInitArgs args;
+  args.backends |= perfetto::kInProcessBackend;
+  perfetto::Tracing::Initialize(args);
+  workerd::TrackEvent::Register();
```

Comment:
I think we'll need a code path somewhere which does
```cpp
if (perfetto::Tracing::IsInitialized()) {
  workerd::TrackEvent::Register();
}
```
The reason is that we need to register both workerd's and our internal codebase's TrackEvents individually, but we can (or maybe just "should"?) only call `perfetto::Tracing::Initialize()` once.